### DISCUSSION TITLE
Mixed-plugins for actuators and viewers

### DISCUSSION THREAD

**Discussion by Ashwolaa** [PyMoDAQ/PyMoDAQ#317] [Ideas]
Hi everyone,

I am currently developing a plugin for a LakeShore Controller (model 340). 

This apparel consists in: 
- a viewing part which effectively gives the temperature of up to four sensors
- a resistive heating part which will deliver a current to heat up some parts in the setup

Currently, I am able to connect with it through a serial connection and I am using some parts of the pymeasure library

Intially, I planned to make a DAQ_move giving it as an input a temperature setpoint to reach and monitoring the temperature evolution to decide if the motion has been achieved or not.

However, I though that it would also be interesting to use it also as viewer to measure the temperature in real time.

In conclusion, I would like to use it both as an actuator and a viewer but I have a single connection. I believe this type of approach could be generalized to many instruments and I would be interested to know your thoughts on the subject and if you have maybe some ideas on how to deal with it

**Comment by seb5g** [comment]
Sure you can do it easily with the concept of master/slave that is also generalised for any control module that can share the same controller, the pymeasure driver in your case

**Comment by rgeneaux** [comment]
Yes, one good example is the Keithley by NTappy which is an amperemeter and a voltage source at the same time!
[https://github.com/LMSC-NTappy/pymodaq_plugins_keithley/](https://github.com/LMSC-NTappy/pymodaq_plugins_keithley/)

**Comment by Ashwolaa** [comment]
Thanks for the fast answers, I will look through the example and the master/slave definition
